lib/house.rb

```diff
@@ -0,0 +1,62 @@
+
+class House
+
+  attr_reader :address, :price, :rooms, :details, :rooms_by_category
+
+  def initialize(price_parameter, address_parameter)
+    @price = price_parameter.delete_prefix('$').to_i
+    @address = address_parameter
+    @rooms = []
+    @details = {"price" => @price, "address" => @address}
```

Comment:
Since we are already storing `@price` and `@address`, we don't need to store the details on the object, this could be a standalone method to cut down on memory used.